File: snippets/go/receive_onchain.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-go/breez_sdk"
)

func GenerateReceiveOnchainAddress() error {
	// ANCHOR: generate-receive-onchain-address
	swapInfo, err := sdk.ReceiveOnchain(breez_sdk.ReceiveOnchainRequest{})
	if err != nil {
		return err
	}
	// Send your funds to the below bitcoin address
	address := swapInfo.BitcoinAddress
	log.Printf("%v", address)

	log.Printf("Minimum amount allowed to deposit in sats: %v", swapInfo.MinAllowedDeposit)
	log.Printf("Maximum amount allowed to deposit in sats: %v", swapInfo.MaxAllowedDeposit)
	// ANCHOR_END: generate-receive-onchain-address
	return nil
}

func GetInProgressSwap() error {
	// ANCHOR: in-progress-swap
	swapInfo, err := sdk.InProgressSwap()
	if err != nil {
		return err
	}
	log.Printf("%#v", swapInfo)
	// ANCHOR_END: in-progress-swap
	return nil
}

func ListRefundables() error {
	// ANCHOR: list-refundables
	refundables, err := sdk.ListRefundables()
	if err != nil {
		return err
	}
	log.Printf("%#v", refundables)
	// ANCHOR_END: list-refundables
	return nil
}

func ExecuteRefund() error {
	// ANCHOR: execute-refund
	refundables, err := sdk.ListRefundables()
	if err != nil {
		return err
	}
	destinationAddress := "..."
	satPerVbyte := uint32(5)
	refundRequest := breez_sdk.RefundRequest{
		SwapAddress: refundables[0].BitcoinAddress,
		ToAddress:   destinationAddress,
		SatPerVbyte: satPerVbyte,
	}
	result, err := sdk.Refund(refundRequest)
	if err != nil {
		return err
	}
	log.Printf("%v", result)
	// ANCHOR_END: execute-refund
	return nil
}

func GetChannelOpeningFees(amountMsat *uint64) error {
	// ANCHOR: get-channel-opening-fees
	channelFees, err := sdk.OpenChannelFee(breez_sdk.OpenChannelFeeRequest{AmountMsat: amountMsat})
	if err != nil {
		return err
	}
	log.Printf("%#v", channelFees)
	// ANCHOR_END: get-channel-opening-fees
	return nil
}

func RescanSwaps() error {
	// ANCHOR: rescan-swaps
	err := sdk.RescanSwaps()
	if err != nil {
		return err
	}
	log.Println("Rescan finished")
	// ANCHOR_END: rescan-swaps
	return nil
}

File: snippets/python/src/receive_onchain.py
import breez_sdk

def generate_receive_onchain_address(sdk_services):
  try: 
    # ANCHOR: generate-receive-onchain-address
    swap_info = sdk_services.receive_onchain(
        req=breez_sdk.ReceiveOnchainRequest())

    # Send your funds to the below bitcoin address
    address = swap_info.bitcoin_address
    print("Minimum amount allowed to deposit in sats:", swap_info.min_allowed_deposit)
    print("Maximum amount allowed to deposit in sats:", swap_info.max_allowed_deposit)
    # ANCHOR_END: generate-receive-onchain-address
  except Exception as error:
        print(error)
        raise
  
def get_in_progress_swap(sdk_services):
  try:
      # ANCHOR: in-progress-swap
      swap_info = sdk_services.in_progress_swap()
      # ANCHOR_END: in-progress-swap
  except Exception as error:
      print(error)
      raise
     
    
def list_refundables(sdk_services):
    try:
        # ANCHOR: list-refundables
        refundables = sdk_services.list_refundables()
        # ANCHOR_END: list-refundables
        return refundables
    except Exception as error:
        print(error)
        raise


def execute_refund(sdk_services, refundable):
    if refundable is breez_sdk.SwapInfo:
        # ANCHOR: execute-refund
        destination_address = "..."
        sat_per_vbyte = 5
        try:
            req = breez_sdk.RefundRequest(
                swap_address=refundable.bitcoin_address,
                to_address=destination_address,
                sat_per_vbyte=sat_per_vbyte)
            result = sdk_services.refund(req=req)
        # ANCHOR_END: execute-refund
        except Exception as error:
            print(error)
            raise


def get_channel_opening_fees(sdk_services, amount_msat=None):
   try:
    # ANCHOR: get-channel-opening-fees
    req = breez_sdk.OpenChannelFeeRequest(amount_msat=amount_msat)
    channel_fees = sdk_services.open_channel_fee(req=req)
    # ANCHOR_END: get-channel-opening-fees
    return channel_fees
   except Exception as error:
       print(error)
       raise


def rescan_swaps(sdk_services):
    try:
        # ANCHOR: rescan-swaps
        sdk_services.rescan_swaps()
        # ANCHOR_END: rescan-swaps        
    except Exception as error:
        print(error)
        raise